conwayconsole/Program.cs

```diff
@@ -9,140 +9,189 @@
+                            if (delay > TimeSpan.Zero)
+                            {
+                                if (delay > TimeSpan.Zero)
+                                {
+                                    await Task.Delay(delay, cts.Token);
+                                }
+
+                                lastLoopTime = DateTime.UtcNow;
+                            }
+                        }
+
+                        if (save)
+                        {
+                            await Save(board);
+                            save = false;
+                            break;
+                        }
+                    }
+                }
+                catch (OperationCanceledException)
+                {
+                }
+                finally
+                {
+                    if (!options.HideDisplay)
+                    {
+                        Console.CursorVisible = true;
+                        Console.ResetColor();
+                        Console.CancelKeyPress -= HandleCancel;
+                    }
+                }
+
+                void HandleCancel(object sender, ConsoleCancelEventArgs args) => cts.Cancel();
             }
-          }
         }
-        catch (OperationCanceledException)
+
+        private static async Task<bool> YesNo(string prompt = null)
         {
+          await Console.Out.WriteAsync(prompt);
+          string response = (await Console.In.ReadLineAsync()).ToLower();
+          return response == "y" || response == "yes";
         }
-        finally
-        {       
-          if (!options.HideDisplay)
-          { 
-            Console.CursorVisible = true;
-            Console.ResetColor();
-            Console.CancelKeyPress -= HandleCancel;
-          }
-        }
+        private static async Task Save(IReadableBoard boardToSave)
+        {
+            if (boardToSave != null)
+            {
+                await Console.Out.WriteAsync("Enter file path (leave blank for default): ");
+                string pathway = await Console.In.ReadLineAsync();
 
-        void HandleCancel(object sender, ConsoleCancelEventArgs args) => cts.Cancel();
-      }
+                bool pathwayIsNullOrEmpty = string.IsNullOrEmpty(pathway);
+                
+                if (!pathwayIsNullOrEmpty)
+                {
+                    pathway = Path.ChangeExtension(pathway, ".json");
+                    string dir = Path.GetDirectoryName(pathway);
+                    if (dir != null && !Directory.Exists(dir))
+                    {
+                        if (await YesNo("Do you want to make a new directory [Y/N]: "))
+                        {
+                            Directory.CreateDirectory(dir);
+                        }
+                        else
+                        {
+                            return;
+                        }
+                    }
+                }
+
+                if (pathwayIsNullOrEmpty)
```

Comment:
Couldn't this just be an `else` on the `if` above? And then the variable `pathwayIsNullOrEmpty` is not needed, as you could just put the `string.IsNullOrEmpty()` inside the `if`.